modules/app-gateway-module-backend/main.tf

```diff
@@ -77,8 +77,8 @@ resource "azurerm_application_gateway" "ag" {
     for_each = [for app in local.gateways[count.index].app_configuration : {
       name          = "${app.product}-${app.component}"
       path          = lookup(app, "health_path_override", "/health/liveness")
-      host_name     = "${app.product}-${app.component}-${var.env}.${local.gateways[count.index].gateway_configuration.host_name_suffix}"
-      ssl_host_name = "${app.product}-${app.component}.${local.gateways[count.index].gateway_configuration.ssl_host_name_suffix}"
+      host_name     = "lookup(app, 'host_name_prefix', ${app.product}-${app.component}-${var.env}).${local.gateways[count.index].gateway_configuration.host_name_suffix}"
```

Comment:
isn't this a host_name_override? not a prefix?

Reply:
hmm, used it as we used host_name_suffix here
 https://github.com/hmcts/azure-platform-terraform/blob/master/environments/sbox/backend_lb_config.yaml#L5 . and we still use it as a suffix after this. May be that should be domain_name?

Reply:
where are you using it as a suffix? not sure if i'm missing something but it looks like `host_name_prefix` is the whole domain name

Reply:
i missed the )